src/recipes/editor-plugin/inspector-plugins.md

```diff
@@ -0,0 +1,342 @@
+ERROR: Cannot get class 'RandomInspectorPlugin'.
+   at: (core/object/class_db.cpp:392)
+ERROR: Cannot get class 'RandomInspectorPlugin'.
+   at: (core/object/class_db.cpp:392)
+```
+
+
+# Full code
+
+
+```rust
+// file: addon.rs
+
+use godot::{
+    engine::{
+        Button, EditorInspectorPlugin, EditorPlugin, EditorProperty, IEditorInspectorPlugin,
+        IEditorPlugin, IEditorProperty,
+    },
+    global,
+    prelude::*,
+};
+use rand::Rng;
+
+#[derive(GodotClass)]
+#[class(tool, init, editor_plugin, base=EditorPlugin)]
+struct RustEditorPlugin {
+    base: Base<EditorPlugin>,
+    random_inspector: Gd<RandomInspectorPlugin>,
+}
+
+#[godot_api]
+impl IEditorPlugin for RustEditorPlugin {
+    fn enter_tree(&mut self) {
+        self.random_inspector = RandomInspectorPlugin::new_gd();
+        self.to_gd()
+            .add_inspector_plugin(self.random_inspector.to_variant().to());
+    }
+
+    fn exit_tree(&mut self) {
+        self.to_gd()
+            .remove_inspector_plugin(self.random_inspector.to_variant().to());
+    }
+}
+
+#[derive(GodotClass)]
+#[class(tool, init, base=EditorInspectorPlugin)]
+struct RandomInspectorPlugin {
+    base: Base<EditorInspectorPlugin>,
+}
+
+#[godot_api]
+impl IEditorInspectorPlugin for RandomInspectorPlugin {
+    fn parse_property(
+        &mut self,
+        _object: Gd<Object>,
+        value_type: VariantType,
+        name: GString,
+        _hint_type: global::PropertyHint,
+        _hit_name: GString,
+        _flags: global::PropertyUsageFlags,
+        _wide: bool,
+    ) -> bool {
+        if value_type == VariantType::INT {
+            self.base_mut()
+                .add_property_editor(name, RandomIntEditor::new_alloc().to_variant().to());
+            return true;
+        }
+
+        false
+    }
+
+    fn can_handle(&self, _object: Gd<Object>) -> bool {
+        true
+    }
+}
+
+#[derive(GodotClass)]
+#[class(tool, init, base=EditorProperty)]
+struct RandomIntEditor {
+    base: Base<EditorProperty>,
+    button: Option<Gd<Button>>,
+}
+
+#[godot_api]
+impl RandomIntEditor {
+    #[func]
+    fn handle_press(&mut self) {
+        let property_name = self.base().get_edited_property();
+        let num = rand::thread_rng().gen_range(0..100);
+        godot_print!("Randomize! {} for {}", num, property_name);
+        self.base_mut()
+            .emit_changed(property_name, num.to_variant().to());
+        if let Some(mut button) = self.button.clone() {
+            let text = format!("Randomize: {}", num);
+            button.set_text(text.into());
+        } else {
+            godot_error!("Button wasn't found in handle_press");
+        }
+    }
+}
+
+#[godot_api]
+impl IEditorProperty for RandomIntEditor {
+    fn enter_tree(&mut self) {
+        let mut button = Button::new_alloc();
+        button.connect(
+            "pressed".into(),
+            self.base().callable("handle_press".to_godot()),
```

Comment:
```suggestion
            self.base().callable("handle_press"),
```
Is this working? Because `base` method exists on `WithBaseField`...

Reply:
yes, this works